### ISSUE TITLE
postmap to generate db

### ISSUE THREAD

**Issue by ploeghvander** [wwkimball/wwkimball-postfix#2]
Hi,
 It's possible to define a postfix lookup file for instance;

/etc/postfix/transport

But the module should also use the "postmap" command to generate a database. Without the postmap command postfix is not able to use the hash lookup function.

Thanks

**Comment by wwkimball** [comment]
I agree this would be nice to have.  There is at least one challenge to this:  Running commands on the host -- say, with an `exec` resource -- outside of a custom Type/Provider is often not portable.  I will take a look at the new Tasks mechanism as a possible solution, but I haven't yet taken the time to play with it.  I don't personally use postmap anywhere in my mail infrastructure, so my personal motivation is admittedly low.

As such, I'll mark this as an enhancement request.  If you or anyone else would like to take a stab at adding this feature, I'd love to see your Pull Requests!  Just please, don't use an `exec` resource for this.

**Comment by ploeghvander** [comment]
yes, at our company we are just starting with puppet. Allthough I already have some experience with puppet at other companies. Exec are the worst yes. The new task folder could indeed be a good option.
Once we are really starting to develop I expect to do modify some stuff and do an pull request. Thanks!
